firestore-translate-text/functions/src/index.ts

```diff
@@ -118,36 +118,56 @@ const handleUpdateDocument = async (
   before: admin.firestore.DocumentSnapshot,
   after: admin.firestore.DocumentSnapshot
 ): Promise<void> => {
-  const inputAfter = extractInput(after);
   const inputBefore = extractInput(before);
+  const inputAfter = extractInput(after);
 
-  const inputHasChanged = inputAfter !== inputBefore;
+  // If previous and updated documents have no input, skip.
+  if (inputBefore === undefined && inputAfter === undefined) {
+    logs.documentUpdatedNoInput();
+    return;
+  }
 
-  if (
-    !inputHasChanged &&
-    inputAfter !== undefined &&
-    inputBefore !== undefined
-  ) {
-    logs.documentUpdatedUnchangedInput();
+  // If updated document has no input, delete any existing translations.
+  if (inputAfter === undefined) {
+    await updateTranslations(after, admin.firestore.FieldValue.delete());
+    logs.documentUpdatedDeletedInput();
     return;
   }
 
-  if (inputAfter) {
+  // If document types do not match, force a translation.
+  if (typeof inputBefore !== typeof inputAfter) {
     logs.documentUpdatedChangedInput();
     await translateDocument(after);
-  } else if (inputBefore) {
-    logs.documentUpdatedDeletedInput();
-    await updateTranslations(after, admin.firestore.FieldValue.delete());
+    return;
+  }
+
+  let beforeValues = [];
```

Comment:
I think you can remove lines 144-157. Stringifying the input & checking for equality should be enough. Update line 159. i.e. `JSON.stringify(inputBefore) === JSON.stringify(inputAfter)`

Reply:
I think you're right here. Will update and retest